docs/sources/reference/commandline/cli.md

```diff
@@ -533,6 +533,10 @@ considered as normal files. Filepaths in .dockerignore are absolute with
 the current directory as the root. Wildcards are allowed but the search
 is not recursive.
 
+When the `URL` references a Dockerfile, or a Dockerfile is read from 
+`STDIN`, then the `-f` option will have no effect and the contents of the
+Dockerfile will be placed in a file called `Dockerfile`.
+
```

Comment:
@moxiegirl did you really want me to add all of that text into the remote api doc?  That looks like its taken from the CLI docs.  Given how terse the rest of this API doc is, I think adding this much text here might be a bit out of place.